prometheus/templates/app_alert.rules.tmpl

```diff
@@ -0,0 +1,96 @@
+groups:
+- name: High CPU
+  rules:
+%{ for instance in apps ~}
+  - alert: High-CPU-${instance.app_name}
+    expr: avg by ( app ) (cpu{app=~"${instance.app_name}"}) > ${instance.max_cpu}
+    for: 5m
+    annotations:
+      summary:     ${instance.app_name} High CPU Alert
+      dashboard:   ${cfapps_dashboard_url}&var-Applications=${instance.app_name}
+      description: "CPU usage has increased in the last 5 minutes (current value: {{ $value }}%)"
+    labels:
+      environment: ${instance.monitoring_space_name}
+      severity:    high
+      app:         ${instance.app_name}
+%{ endfor ~}
+
+- name: Memory Utilisation
+  rules:
+%{ for instance in apps ~}
+  - alert: High-Memory-Utilisation-${instance.app_name}
+    expr: avg by ( app ) (memory_utilization{app=~"${instance.app_name}"}) > ${instance.max_mem}
+    for: 5m
+    annotations:
+      summary:     ${instance.app_name} high memory utilization
+      dashboard:   ${cfapps_dashboard_url}&var-Applications=${instance.app_name}
+      description: "Memory utilization has increased in the last 5 minutes (current value: {{ $value }}%)"
+    labels:
+      severity:    high
+      app:         ${instance.app_name}
+      environment: ${instance.monitoring_space_name}
+%{ endfor ~}
+
+- name: Disk Utilisation
+  rules:
+%{ for instance in apps ~}
+  - alert: High-Disk-Utilisation-${instance.app_name}
+    expr: avg by ( app ) ( disk_utilization{ app=~"${instance.app_name}" }) > ${instance.max_disk}
+    for: 5m
+    annotations:
+      summary:     ${instance.app_name} high disk utilization
+      dashboard:   ${cfapps_dashboard_url}&var-Applications=${instance.app_name}
+      description: "Disk utilization has increased in the last 5 minutes (current value: {{ $value }})%"
+    labels:
+      severity:    high
+      app:         ${instance.app_name}
+      environment: ${instance.monitoring_space_name}
+%{ endfor ~}
+
+- name: App Crashes
+  rules:
+%{ for instance in apps ~}
+  - alert: App-Crash-${instance.app_name}
+    expr: rate(crash{app=~"${instance.app_name}"}[1m])*60 > ${instance.max_crash_count}
+    for: 5m
+    annotations:
+      summary:     At least one instance of ${instance.app_name} has crashed in the last 5 mins
```

Comment:
There may be more than one if max_crash_count > 1

Reply:
@saliceti  so what's the fix for this ?